### PR TITLE
Support DF for subsequence.

### PR DESCRIPTION
**Pull request by ghislainfourny**

### CODE REVIEW

File: src/main/java/org/rumbledb/runtime/functions/sequences/general/SubsequenceFunctionIterator.java [hunk 1]
```diff
@@ -79,6 +86,53 @@ public SubsequenceFunctionIterator(
...
+        df.createOrReplaceTempView("input");
+        df = df.sparkSession()
+            .sql(
+                String.format(
+                    "SELECT * FROM input LIMIT %s",
+                    Integer.toString(this.startPosition + this.length - 1)
+                )
+            );
+
+        df = FlworDataFrameUtils.zipWithIndex(df, 1L, "foo");
```
Comment (CanBerker) [inline src/main/java/org/rumbledb/runtime/functions/sequences/general/SubsequenceFunctionIterator.java]: "foo" is very likely to collide with a user column. Can we use a UUID for the column name here?

Comment (ghislainfourny) [reply src/main/java/org/rumbledb/runtime/functions/sequences/general/SubsequenceFunctionIterator.java]: Ah, yes! That was the plan and I forgot. Thanks! Very good eyes.

**Comment by CanBerker** [comment]
This is optional but shall we also add tests for the behavior at the edge values ? Lower than first index, bigger than last index etc.
